pulp_rpm/app/tasks/publishing.py

```diff
@@ -199,9 +200,42 @@ def populate(publication):
         packages (pulp_rpm.models.Package): A list of published packages.
 
     """
-    packages = Package.objects.filter(pk__in=publication.repository_version.content).\
-        prefetch_related('contentartifact_set')
     published_artifacts = []
+    contents = publication.repository_version.content
+
+    distribution_trees = DistributionTree.objects.filter(
+        pk__in=publication.repository_version.content
+    ).prefetch_related(
+        "addons", "variants", "addons__repository", "variants__repository", "contentartifact_set"
+    )
+
+    for distribution_tree in distribution_trees:
+        for content_artifact in distribution_tree.contentartifact_set.all():
+            published_artifacts.append(PublishedArtifact(
+                relative_path=content_artifact.relative_path,
+                publication=publication,
+                content_artifact=content_artifact)
+            )
+        for addon in distribution_tree.addons.all():
+            repository_version = RepositoryVersion.latest(addon.repository)
+            if repository_version and repository_version.content != contents:
```

Comment:
What is `repository_version.content != contents` checking for?

Reply:
`contents` is referred to contents on main repo, in some cases `variants` points to main repo. So it is checking if the sub-repos(variants/addons) are not the same from the main repo.